.github/workflows/tests.yml

```diff
@@ -147,7 +147,7 @@ jobs:
         run: |
           python -m pip install --upgrade pip
           python -m pip install accelerate==0.34.0
-          python -m pip install datasets==2.21.0
```

Comment:
This test would fail with 2.21. So we bump to the next version: 3.0.0:

```FAILED tests/test_sft_trainer.py::SFTTrainerTester2::test_train_with_iterable_dataset - ValueError: Batch does not contain any data (`None`). At the end of all iterable data available before expected stop iteration.```